### PR TITLE
Denops v6

### PR DESCRIPTION
**Pull request by lambdalisue**
## For Users

Changes seen by users using the Denops plugin are mostly limited to updates of supported versions, with no significant changes other than that.
### Updates of Supported Versions

In Denops v6, the following versions are supported:

| Name | Version |
| ---- | ---- |
| Deno | 1.38.5 or later |
| Vim | 9.0.2189 or later |
| Neovim | 0.9.4 or later |
### Approximately 50% Reduction in Memory Usage

The memory usage of the Denops server has been reduced by approximately half compared to v5. For more details, refer to [Memory (RSS) usage difference between denops v5 and v6](https://github.com/vim-denops/denops.vim/wiki/Memory-(RSS)-usage-difference-between-denops-v5-and-v6).

### Improved Communication Speed with Vim/Neovim

Communication speed with Vim/Neovim has been enhanced, especially noticeable in Neovim, where under specific conditions, you can issue about 1.8 times more commands compared to v5. For detailed results, see [Communication performance difference between denops v5 and v6](https://github.com/vim-denops/denops.vim/wiki/Communication-performance-difference-between-denops-v5-and-v6).

### Reduced Impact on Initialization Time Due to the Number of Plugins

The issue of increasing initialization time as the number of plugins dependent on Denops grows has been resolved. However, users who do not utilize a large number (around 100) of Denops plugins will not perceive a significant difference. For detailed results, refer to [Startup performance difference between v5 and v6](https://github.com/vim-denops/denops.vim/wiki/Startup-performance-difference-between-denops-v5-and-v6).

### Plugins are no longer isolated by Worker threads

Until v5, each plugin was processed in a dedicated Worker thread, but from v6 onwards, processing is done in a Worker thread per Vim/Neovim instance. Be cautious if your code uses global variables, and consider using the `denops.context (Record<PropertyKey, unknown>)` provided for each plugin instead.

### Results of `console.log()` always returned to Vim/Neovim

Until v5, the output location of `console.log()` differed between Local server and Shared server, but now it is always displayed on the Vim/Neovim side in both cases.

> [!TIP]
> `console.log()` is intended for temporary use during development. If you want to display messages to users as part of the plugin's functionality, continue to use `denops.cmd("echo ...")` or the `echo` function in `denops_std`.

### `denops_core` has been separated into a separate repository

The type information of Denops, which was managed in the `denops/@denops` directory until v5, has been separated into a separate repository called [deno-denops-core](https://github.com/vim-denops/deno-denops-core). With this change, the version update policy of "no releases other than updates under the `denops/@denops` directory" has been abolished, and version updates will follow semantic versioning.

### Lower-level test utilities included, separate from `denops_test`

A module named `denops/@denops-private/testutil` has been added to directly test the `Host` used in the internal implementation of Denops. As a result, the parts that were tested using `denops_test` have been removed, and testing is now self-contained in `denops.vim`.

## Summary by CodeRabbit

- **New Features**
  - Introduced a new logging function to improve information display.
  - Added new test utilities to enhance plugin development and testing.
  - Implemented a new worker-based communication approach for better performance.

- **Enhancements**
  - Upgraded support for Deno, Vim, and Neovim to their latest versions.
  - Improved error serialization and deserialization for robust error handling.
  - Refined the communication protocol between services and hosts (Vim/Neovim).

- **Bug Fixes**
  - Fixed issues with autocmd event handling by removing outdated events.
  - Resolved version check discrepancies in the Vim plugin.

- **Refactor**
  - Streamlined tasks and commands in workflow and configuration files.
  - Overhauled internal class structures and method calls for clarity and efficiency.

- **Documentation**
  - Updated README.md with current versioning information and streamlined content.

- **Chores**
  - Cleaned up obsolete functionality and updated import statements for consistency.

- **Tests**
  - Added comprehensive tests for new and existing functionalities to ensure stability.

### CODE REVIEW

**Comment by Shougo** [comment]
I have tested it in Windows environment.
The initialization time is longer than previous version.
It tooks time 5000ms+.

File: denops/@denops-private/host/vim.ts [hunk 1]
```diff
@@ -1,30 +1,15 @@
...
-  ] as const,
-);
+} from "https://deno.land/x/vim_channel_command@v3.0.0/mod.ts";
+import { formatCall, Host, invoke, Service } from "../host.ts";
 
 export class Vim implements Host {
   #session: Session;
   #client: Client;
-  #invoker?: Invoker;
+  #service?: Service;
```
[the remark this answers is not in the document]
Comment (lambdalisue) [reply denops/@denops-private/host/vim.ts:12]: What the difference?

[the remark this answers is not in the document]
Comment (lambdalisue) [reply denops/@denops-private/host/vim.ts:12]: Well, we won't allow developers to assign undefined to #service. We use #service? here to express that the #service might not be initialized until init() is invoked.
Even with this purpose, should we still apply your suggestion?

**Comment by Shougo** [comment]
It works for me.

**Pull request merged**